### PR TITLE
add node require hook

### PR DESCRIPTION
**Pull request by andreypopp**

### CODE REVIEW

File: register.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+var removeTypes = require('./index');
+
+var prev = require.extensions['.js'];
+
+require.extensions['.js'] = function(module, filename) {
+  if (filename.indexOf('node_modules') === -1) {
+    var source = fs.readFileSync(filename, 'utf8');
+    module._compile(removeTypes(source), filename);
+  } else {
+    prev(module, filename);
```
Comment (leebyron) [inline register.js]: prev is undefined

Comment (andreypopp) [inline register.js]: Interesting, I tested it and it's fine with Node 5. Might be a different node version then...

Comment (leebyron) [inline register.js]: Oh never mind. I just missed it

**Comment by leebyron** [comment]
This is so good! Could you this to the npm test plan please?

**Comment by leebyron** [comment]
For a test, you could just write a dead simple file which includes some flow definitions and just run it with node from npm test.

**Comment by andreypopp** [comment]
done

**Comment by andreypopp** [comment]
also added case for `prev` being not defined

**Comment by leebyron** [comment]
Awesome!

**Pull request merged**